react/base.yml

```diff
@@ -0,0 +1,57 @@
+# Enforce best practices and protect against common mistakes in JSX and React.
+#
+# This is an alternative to openstax-poland/base which includes additional rules
+# from eslint-plugin-react.
+
+extends:
+  - ../base.yml
+  - plugin:react/recommended
+
+settings:
+  react:
+    version: 16.2.0
+
+rules:
+  # Forbid use of .bind() and arrow functions as JSX callbacks, as they are not
+  # comparable and can cause unnecessary re-renders.
+  react/jsx-no-bind: error
+
+  # Forbid uses of fragments with a single child element.
+  react/jsx-no-useless-fragment: error
+
+  # Don't access this.state within this.setState.
+  react/no-access-state-in-setstate: error
+
+  # Disallow this.setState within componentDidMount and componentDidUpdate,
+  # as this will trigger an additional render.
+  react/no-did-mount-set-state: error
+  react/no-did-update-set-state: error
```

Comment:
This will complicate components even more and since we can use react hooks I think those rules should be set to `warn`